packages/compass/src/app/index.tsx

```diff
@@ -91,6 +94,59 @@ import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 import { getAppName, getAppVersion } from '@mongodb-js/compass-utils';
 const { log, mongoLogId, track } = createLoggerAndTelemetry('COMPASS-APP');
 
+const WithPreferencesAndLoggerProviders: React.FC = ({ children }) => {
+  const loggerProviderValue = {
+    createLogger: createLoggerAndTelemetry,
+    preferences: defaultPreferencesInstance,
+  };
+  return (
+    <PreferencesProvider value={loggerProviderValue.preferences}>
+      <LoggerAndTelemetryProvider value={loggerProviderValue}>
+        {children}
+      </LoggerAndTelemetryProvider>
+    </PreferencesProvider>
+  );
+};
+
+const WithAtlasProviders: React.FC = ({ children }) => {
+  return (
+    <AtlasAuthServiceProvider value={new CompassAtlasAuthService()}>
+      <AtlasServiceProvider
+        options={{
+          defaultHeaders: {
+            'User-Agent': `${getAppName()}/${getAppVersion()}`,
+          },
+        }}
+      >
+        <AtlasAiServiceProvider>{children}</AtlasAiServiceProvider>
+      </AtlasServiceProvider>
+    </AtlasAuthServiceProvider>
+  );
+};
+
+const WithStorageProviders: React.FC = ({ children }) => {
+  const favoriteQueryStorageProviderValue: FavoriteQueryStorageAccess = {
+    getStorage(options) {
+      return new CompassFavoriteQueryStorage(options);
+    },
+  };
+  const recentQueryStorageProviderValue: RecentQueryStorageAccess = {
+    getStorage(options) {
+      return new CompassRecentQueryStorage(options);
+    },
+  };
+
+  return (
+    <PipelineStorageProvider value={new CompassPipelineStorage()}>
+      <FavoriteQueryStorageProvider value={favoriteQueryStorageProviderValue}>
+        <RecentQueryStorageProvider value={recentQueryStorageProviderValue}>
```

Comment:
Same as above, now that this code is part of component render, let's ensure we don't re-create instances

```suggestion
  const favoriteQueryStorageProviderValue: FavoriteQueryStorageAccess = useRef({
    getStorage(options) {
      return new CompassFavoriteQueryStorage(options);
    },
  });
  const recentQueryStorageProviderValue: RecentQueryStorageAccess = useRef({
    getStorage(options) {
      return new CompassRecentQueryStorage(options);
    },
  });
  const pipelineStorage = useRef(new CompassPipelineStorage());
  
  return (
    <PipelineStorageProvider value={pipelineStorage.current}>
      <FavoriteQueryStorageProvider value={favoriteQueryStorageProviderValue.current}>
        <RecentQueryStorageProvider value={recentQueryStorageProviderValue.current}>
```